filters/auth/jwt_validation.go

```diff
@@ -0,0 +1,301 @@
+package auth
+
+import (
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-co-op/gocron"
+	"github.com/golang-jwt/jwt"
+	"github.com/opentracing/opentracing-go"
+	log "github.com/sirupsen/logrus"
+	"github.com/zalando/skipper/filters"
+)
+
+const (
+	JwtValidationName = "jwtValidation"
+)
+
+type (
+	jwtValidationSpec struct {
+		options TokenintrospectionOptions
+	}
+
+	jwtValidationFilter struct {
+		scheduler       *gocron.Scheduler
+		claims          []string
+		upstreamHeaders map[string]string
+	}
+)
+
+var m sync.Mutex
+var rsakeys map[string]*rsa.PublicKey = make(map[string]*rsa.PublicKey)
+
+func NewJwtValidationWithOptions(o TokenintrospectionOptions) filters.Spec {
+	return &jwtValidationSpec{
+		options: o,
+	}
+}
+
+func NewJwtValidation(timeout time.Duration) filters.Spec {
+	return NewJwtValidationWithOptions(TokenintrospectionOptions{
+		Timeout: timeout,
+		Tracer:  opentracing.NoopTracer{},
+	})
+}
+
+func JwtValidationWithOptions(
+	create func(time.Duration) filters.Spec,
+	o TokenintrospectionOptions,
+) filters.Spec {
+	s := create(o.Timeout)
+	ts, ok := s.(*jwtValidationSpec)
+	if !ok {
+		return s
+	}
+
+	ts.options = o
+	return ts
+}
+
+func (s *jwtValidationSpec) Name() string {
+	return JwtValidationName
+}
+
+func (s *jwtValidationSpec) CreateFilter(args []interface{}) (filters.Filter, error) {
+	sargs, err := getStrings(args)
+	if err != nil {
+		return nil, err
+	}
+	if len(sargs) < 2 {
+		return nil, filters.ErrInvalidFilterParameters
+	}
+
+	issuerURL := sargs[0]
+
+	cfg, err := getOpenIDConfig(issuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	scheduler := gocron.NewScheduler(time.UTC)
+	scheduler.Every("30m").Do(func() {
+		err = fillPublicKeys(cfg.JwksURI)
+		if err != nil {
+			log.Errorf("Error getting public keys for url: %s", cfg.JwksURI)
+			return
+		}
+	})
+
+	scheduler.StartAsync()
```

Comment:
Lets also consider existing options referenced by https://github.com/golang-jwt/jwt/issues/73

Reply:
Thanks for the hint. Will try to implement approach based on https://github.com/MicahParks/keyfunc